README.rst

```diff
@@ -537,16 +537,47 @@ the run step:
           namespaced.file.alias2: "/path/to/readwrite/file/or/dir:rw"
           build/dir/file: "/path/to/build/dir/file"
 
-        # A map specifying cache directories that should be mounted inside the
-        # container. The cache directories are maintained between builds and can
-        # be used to store files, such as downloaded dependencies, to speed up
-        # builds. Caches are shared within a build configuration, meaning that
-        # caches with the same name are shared between steps. Caches should be
-        # treated as ephemeral and should only store items that can be
-        # obtained/generated by subsequent builds.
+        # A map specifying cache directories that stored as archive files on the
+        # host system as named <local cache key> and extracted as directory in
+        # the container name <docker path>. The cache directories are maintained
+        # between builds and can be used to store files, such as downloaded
+        # dependencies, to speed up builds. Caches are shared within a build
+        # configuration, meaning that caches with the same name are shared between
+        # steps. Caches should be treated as ephemeral and should only store items
+        # that can be obtained/generated by subsequent builds.
+        #
+        # Two formats are supported when defining caches.
+        # 1) <local cache key>: <docker path> (backwards compatible)
+        #
+        # 2) <docker path>:
+        #      - <local cache key A>
+        #      - <local cache key B>
```

Comment:
I would reformat this a bit to switch around #1 and #2 so that #2 is put first and also put a "RECOMMENDED" label or something on it. The backwards compatible one should probably say `<local cache key>: <docker path> (backwards compatible with older caching method, but more limited)` since it won't be clear what it's backwards compatible with.